### PR TITLE
fix: convert column type to str during dual read/write

### PR DESCRIPTION
**Pull request by hughhhh**
### SUMMARY

Users are blocked when trying to save a dataset on a query/table that has an ARRAY column in it. Due the python_type returning a list instead of a str for the type. To fix this i've added a util function to check if the type is a list and convert it to a string like this `ARRAY`

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

**Comment by eschutho** [comment]
This is great @hughhhh. Can you write a quick unit test for it?

**Comment by ktmud** [comment]
should we save the original database type here?

What does the old column model do?

**Comment by hughhhh** [comment]
> Should we save the original database type here? The type field specifically used Text as storage so the originally intention is probably to allow storing the full schema of complex types such as ARRAY and STRUCT?
> 
> What does the old column model save?

I'm pretty sure it was still ARRAY, but this is only a patch i'm pretty sure there other types we'd have to map to strings later

File: superset/connectors/sqla/utils.py [hunk 1]
```diff
@@ -180,6 +180,15 @@ def is_column_type_temporal(column_type: TypeEngine) -> bool:
         return False
 
 
+def convert_column_type(column_type: TypeEngine) -> str:
+    if column_type.python_type == list:
+        return "ARRAY"
+    if column_type.python_type == dict:
+        return "JSON"
```
Comment (ktmud) [inline superset/connectors/sqla/utils.py:187]: Can it also be a `STRUCT`? I guess it probably depends on the database.

Comment (hughhhh) [reply superset/connectors/sqla/utils.py:187]: so i'm following the pattern from the `sqlalchemy.sql.sqltypes`

Comment (villebro) [reply superset/connectors/sqla/utils.py:187]: > Should we save the original database type here? The type field specifically used Text as storage so the originally intention is probably to allow storing the full schema of complex types such as ARRAY and STRUCT?

I agree, using the original native type is the ideal solution here.

**Comment by villebro** [comment]
@hughhhh @ktmud I've opened up an alternative solution for this issue that reuses existing code that's used when creating physical datasets: #19714

**Comment by sadpandajoe** [comment]
🏷️  preset:2022.15

**Pull request closed without merging**